packages/core/src/inference/function-type.js

```diff
@@ -219,10 +219,18 @@ export function inferenceFunctionLiteralType(
             paramType.variants.every(
               variant =>
                 variant instanceof CollectionType ||
-                variant instanceof TupleType
+                variant instanceof TupleType ||
+                (
+                  variant instanceof $BottomType &&
+                  variant.getRootedSubordinateType().isPrincipalTypeFor(new TypeVar("Array"))
+                )
             )) ||
           paramType instanceof CollectionType ||
-          paramType instanceof TupleType
+          paramType instanceof TupleType ||
+          (
+            paramType instanceof $BottomType &&
+            paramType.getRootedSubordinateType().isPrincipalTypeFor(new TypeVar("Array"))
+          )
```

Comment:
@JSMonk  i am not sure if i am doing it correctly but tests are happy with this...

Reply:
Oh, you need to use `CollectionType.Array` Instead of new TypeVar(“Array”)